src/main/java/de/slimecloud/slimeball/config/commands/MenuCommand.java

```diff
@@ -4,32 +4,506 @@
+	private static MessageMenu createMapValueMenu(@NotNull SlimeBot bot, @NotNull UIManager manager, @NotNull Class<?> type, @NotNull Type generic, @NotNull Function<GuildConfig, Object> categoryInstance, @NotNull Getter getter, @NotNull Setter setter, @NotNull Function<DataState<?>, String> display, @NotNull CategoryInfo category, @NotNull ConfigField field, @Nullable ConfigFieldType keyType) {
+		Class<?> keyClass = getClass(getGenericType(generic, 0));
+
+		Type valueType = getGenericType(generic, 1);
+		Class<?> valueClass = getClass(valueType);
+
+		//Check if type arguments are valid
+		if (Map.class.isAssignableFrom(keyClass) || Collection.class.isAssignableFrom(keyClass)) throw new UnsupportedOperationException();
+		if (Map.class.isAssignableFrom(valueClass)) throw new UnsupportedOperationException("Cannot create config menu with map inside of map");
+
+		List<Component<?>> components = new ArrayList<>();
+
+		//Add base components
+		components.add(new ButtonComponent("back", ButtonColor.GRAY, "Zurück").appendHandler(s -> manager.getMenu("config." + category.command()).display(s.event)));
+		components.add(new ButtonComponent("reset", ButtonColor.RED, "Zurücksetzten").appendHandler(s -> {
+			set(bot, s, categoryInstance, setter, type.isAssignableFrom(HashMap.class) ? new HashMap<>() : new LinkedHashMap<>());
+			s.update();
+		}));
+
+		//Create virtual info environment for sub-menus
+		CategoryInfo valueCategory = createCategory(category.name() + " → " + field.name(), category.command() + "." + field.command(), field.description());
+		ConfigField valueField = createField("?", "value", field.description(), field.type());
+
+		//The value menu shows the current value for a key. This can also be used to change the value
+		MessageMenu valueMenu = createFieldMenu(bot, manager, valueType, categoryInstance,
+				//Implementation for getter method
+				(s, c) -> ((Map) getter.get(s, c)).computeIfAbsent(s.getRawState("key", keyClass), x -> Collection.class.isAssignableFrom(valueClass) ? createEmptyCollection(valueClass, valueType) : null),
+				//Implementation for setter method
+				(s, c, v) -> {
+					Map value = (Map) getter.get(s, c);
+
+					//Read key from state
+					Object key = keyType.parse(keyClass, s.getRawState("key", keyClass));
+
+					if (v != null) value.put(key, v);
+					else value.remove(key);
+				},
+				s -> keyType.toString(s.getRawState("key", keyClass)), valueCategory, valueField, null
+		);
+
+		//Add components that opens the value menu to add a new entry
+		Component<?> add = keyType.createComponent(manager, keyClass, "config." + category.command() + "." + field.command(), "add", "Wert hinzufügen", (s, k) -> {
+			valueMenu.createState(s).setState("key", k).display(s.event);
+			throw new RenderTermination();
+		});
+
+		//Add component to remove th value for an existing key
+		Component<?> remove = new StringSelectComponent("remove", s -> MenuCommand.<Map<?, ?>>get(bot, s, getter).keySet().stream()
```

Comment:
After click -> Allgemein -> "Automatisches Nachrichtenlöschen"
```java
18:18:12 JDA MainWS-ReadThread      EventManager         ERROR  Error in ErrorHandler
java.lang.NullPointerException: Cannot invoke "java.util.Map.keySet()" because the return value of "de.slimecloud.slimeball.config.commands.MenuCommand.get(de.slimecloud.slimeball.main.SlimeBot, de.mineking.discordutils.ui.state.DataState, de.slimecloud.slimeball.config.commands.MenuCommand$Getter)" is null
	at de.slimecloud.slimeball.config.commands.MenuCommand.lambda$createMapValueMenu$46(MenuCommand.java:396)
	at de.mineking.discordutils.ui.components.select.StringSelectComponent.build(StringSelectComponent.java:177)
	at de.mineking.discordutils.ui.MessageMenu.lambda$buildMessage$0(MessageMenu.java:65)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at de.mineking.discordutils.ui.MessageMenu.lambda$buildMessage$1(MessageMenu.java:67)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at de.mineking.discordutils.ui.MessageMenu.buildMessage(MessageMenu.java:69)
	at de.mineking.discordutils.ui.state.MessageSendState.display(MessageSendState.java:56)
	at de.mineking.discordutils.ui.state.MessageSendState.display(MessageSendState.java:72)
	at de.mineking.discordutils.ui.state.MessageSendState.display(MessageSendState.java:25)
	at de.mineking.discordutils.ui.components.button.MenuComponent.lambda$new$1(MenuComponent.java:35)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at de.mineking.discordutils.ui.components.button.MenuComponent.lambda$new$2(MenuComponent.java:33)
	at de.mineking.discordutils.ui.components.button.ButtonComponent.lambda$appendHandler$9(ButtonComponent.java:117)
	at de.mineking.discordutils.ui.components.button.ButtonComponent$1.handleEvent(ButtonComponent.java:213)
	at de.mineking.discordutils.ui.components.button.ButtonComponent$1.handleEvent(ButtonComponent.java:208)
	at de.mineking.discordutils.events.EventHandler.handle(EventHandler.java:37)
	at de.mineking.discordutils.events.EventManager.lambda$onGenericEvent$0(EventManager.java:35)
	at java.base/java.util.Collection.removeIf(Collection.java:576)
	at de.mineking.discordutils.events.EventManager.onGenericEvent(EventManager.java:35)
	at net.dv8tion.jda.api.hooks.ListenerAdapter.onEvent(ListenerAdapter.java:430)
	at net.dv8tion.jda.api.hooks.InterfacedEventManager.handle(InterfacedEventManager.java:96)
	at net.dv8tion.jda.internal.hooks.EventManagerProxy.handleInternally(EventManagerProxy.java:88)
	at net.dv8tion.jda.internal.hooks.EventManagerProxy.handle(EventManagerProxy.java:70)
	at net.dv8tion.jda.internal.JDAImpl.handleEvent(JDAImpl.java:177)
	at net.dv8tion.jda.internal.handle.InteractionCreateHandler.handleAction(InteractionCreateHandler.java:134)
	at net.dv8tion.jda.internal.handle.InteractionCreateHandler.handleInternally(InteractionCreateHandler.java:86)
	at net.dv8tion.jda.internal.handle.SocketHandler.handle(SocketHandler.java:39)
	at net.dv8tion.jda.internal.requests.WebSocketClient.onDispatch(WebSocketClient.java:1015)
	at net.dv8tion.jda.internal.requests.WebSocketClient.onEvent(WebSocketClient.java:901)
	at net.dv8tion.jda.internal.requests.WebSocketClient.handleEvent(WebSocketClient.java:879)
	at net.dv8tion.jda.internal.requests.WebSocketClient.onBinaryMessage(WebSocketClient.java:1054)
	at com.neovisionaries.ws.client.ListenerManager.callOnBinaryMessage(ListenerManager.java:385)
	at com.neovisionaries.ws.client.ReadingThread.callOnBinaryMessage(ReadingThread.java:276)
	at com.neovisionaries.ws.client.ReadingThread.handleBinaryFrame(ReadingThread.java:996)
	at com.neovisionaries.ws.client.ReadingThread.handleFrame(ReadingThread.java:755)
	at com.neovisionaries.ws.client.ReadingThread.main(ReadingThread.java:108)
	at com.neovisionaries.ws.client.ReadingThread.runMain(ReadingThread.java:64)
	at com.neovisionaries.ws.client.WebSocketThread.run(WebSocketThread.java:45)

Reply:
still error at "Automatisches Nachrichtenlöschen"

```java
21:52:46 JDA MainWS-ReadThread      EventManager         ERROR  Error in ErrorHandler
java.lang.NumberFormatException: For input string: "<#1088971160420614237>"
	at java.base/java.lang.NumberFormatException.forInputString(NumberFormatException.java:67)
	at java.base/java.lang.Long.parseLong(Long.java:711)
	at java.base/java.lang.Long.parseLong(Long.java:836)
	at de.slimecloud.slimeball.config.engine.ConfigFieldType$2.parse(ConfigFieldType.java:119)
	at de.slimecloud.slimeball.config.commands.MenuCommand.lambda$createMapValueMenu$43(MenuCommand.java:385)
	at de.slimecloud.slimeball.config.commands.MenuCommand.get(MenuCommand.java:192)
	at de.slimecloud.slimeball.config.commands.MenuCommand.lambda$createEnumSetValueMenu$31(MenuCommand.java:326)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at de.slimecloud.slimeball.config.commands.MenuCommand.lambda$createEnumSetValueMenu$32(MenuCommand.java:328)
	at de.mineking.discordutils.ui.components.select.StringSelectComponent.build(StringSelectComponent.java:177)
	at de.mineking.discordutils.ui.MessageMenu.lambda$buildMessage$0(MessageMenu.java:65)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at de.mineking.discordutils.ui.MessageMenu.lambda$buildMessage$1(MessageMenu.java:67)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575)
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616)
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622)
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627)
	at de.mineking.discordutils.ui.MessageMenu.buildMessage(MessageMenu.java:69)
	at de.mineking.discordutils.ui.state.MessageSendState.display(MessageSendState.java:57)
	at de.mineking.discordutils.ui.state.MessageSendState.display(MessageSendState.java:72)
	at de.slimecloud.slimeball.config.commands.MenuCommand.lambda$createMapValueMenu$47(MenuCommand.java:401)
	at de.slimecloud.slimeball.config.engine.ConfigFieldType$2.lambda$createComponent$0(ConfigFieldType.java:111)
	at de.mineking.discordutils.ui.components.select.EntitySelectComponent.lambda$appendHandler$5(EntitySelectComponent.java:64)
	at de.mineking.discordutils.ui.components.select.EntitySelectComponent$1.handleEvent(EntitySelectComponent.java:184)
	at de.mineking.discordutils.ui.components.select.EntitySelectComponent$1.handleEvent(EntitySelectComponent.java:181)
	at de.mineking.discordutils.events.EventHandler.handle(EventHandler.java:37)
	at de.mineking.discordutils.events.EventManager.lambda$onGenericEvent$0(EventManager.java:35)
	at java.base/java.util.Collection.removeIf(Collection.java:576)
	at de.mineking.discordutils.events.EventManager.onGenericEvent(EventManager.java:35)
	at net.dv8tion.jda.api.hooks.ListenerAdapter.onEvent(ListenerAdapter.java:430)
	at net.dv8tion.jda.api.hooks.InterfacedEventManager.handle(InterfacedEventManager.java:96)
	at net.dv8tion.jda.internal.hooks.EventManagerProxy.handleInternally(EventManagerProxy.java:88)
	at net.dv8tion.jda.internal.hooks.EventManagerProxy.handle(EventManagerProxy.java:70)
	at net.dv8tion.jda.internal.JDAImpl.handleEvent(JDAImpl.java:177)
	at net.dv8tion.jda.internal.handle.InteractionCreateHandler.handleAction(InteractionCreateHandler.java:147)
	at net.dv8tion.jda.internal.handle.InteractionCreateHandler.handleInternally(InteractionCreateHandler.java:86)
	at net.dv8tion.jda.internal.handle.SocketHandler.handle(SocketHandler.java:39)
	at net.dv8tion.jda.internal.requests.WebSocketClient.onDispatch(WebSocketClient.java:1015)
	at net.dv8tion.jda.internal.requests.WebSocketClient.onEvent(WebSocketClient.java:901)
	at net.dv8tion.jda.internal.requests.WebSocketClient.handleEvent(WebSocketClient.java:879)
	at net.dv8tion.jda.internal.requests.WebSocketClient.onBinaryMessage(WebSocketClient.java:1054)
	at com.neovisionaries.ws.client.ListenerManager.callOnBinaryMessage(ListenerManager.java:385)
	at com.neovisionaries.ws.client.ReadingThread.callOnBinaryMessage(ReadingThread.java:276)
	at com.neovisionaries.ws.client.ReadingThread.handleBinaryFrame(ReadingThread.java:996)
	at com.neovisionaries.ws.client.ReadingThread.handleFrame(ReadingThread.java:755)
	at com.neovisionaries.ws.client.ReadingThread.main(ReadingThread.java:108)
	at com.neovisionaries.ws.client.ReadingThread.runMain(ReadingThread.java:64)
	at com.neovisionaries.ws.client.WebSocketThread.run(WebSocketThread.java:45)